### DISCUSSION TITLE
Invalid transaction handle (expecting explicit transaction start)]

### DISCUSSION THREAD

**Discussion by marcin-bury** [asfernandes/node-firebird-drivers#100] [Q&A] [answered]
Hi
It's me again :-)

I have several routines where I pass attachment and transaction in parameters, that I call one by one. At the end of the sequence I try to call transaction.commit() and receive the error 'invalid transaction handle...'.
I have only checked on executing several insert statements (via attachment.execute or attachment.executeReturn), 
I have even add commitRetaining after each execution, but no luck

Any help appreciated very much

Marcin

**Comment by asfernandes** [comment, marked as answer]
Are you using `await` or chaining promises correctly?

**Reply by marcin-bury** [reply]
Thanks for the tip
There where "something" with iterating detail items from the payload via forEach method. Probably my lack of knowledge. I have change it to standard "for" loop and now it works like a charm

Thanks again for the patience.
Marcin
